omg/mechanics/movement.py

```diff
@@ -126,18 +149,21 @@ def calculate_player_state(
         center_x,
         center_y,
         mouse_x,
-        mouse_y,
+        mouse_y, 
         mov_speed_lr,
         mov_speed_ud,
         *args,
         **_ignored
     ):
+        [self.change_direction_x, self.change_direction_y] = (
+            self._calculate_displacement_directions()
+        )
         """Given input player state, move the player.
 
         Velocity is defined relative to the Window.
         """
         vx_wrt_ground = mov_speed_lr * self.change_direction_x
         vy_wrt_ground = mov_speed_ud * self.change_direction_y
-        angle_rad = self.face_mouse(mouse_x, mouse_y, center_x, center_y)
-        angle = math.degrees(angle_rad)
-        return vx_wrt_ground, vy_wrt_ground, angle
+        angle = 0   #Angle is always 0 to enable proper animations
```

Comment:
The problem here is: this `angle` dictates both character_angle` and the `shooting_angle`. 

You wanted to only constrain `char_angle`. That means we need two `angle` variables one defining `character_angle` and the other one `shooting_angle`. The latter can be calculated with the mouse position.